File: iossdkdemo/SDKDemo/APIControllers/GetSurveysViewController.m
//
//  GetSurveysViewController.m
//  SDKDemo
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "GetSurveysViewController.h"
#import <OPGSDK/OPGSDK.h>
#import "SurveyTableViewController.h"

@interface GetSurveysViewController ()

@property(strong, nonatomic) NSArray* surveyList;
@property(strong, nonatomic) IBOutlet UITextField* panelField;

@end

@implementation GetSurveysViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.panelField.delegate=self;
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBAction
-(IBAction)getSurveyObjC:(id)sender
{
    OPGSDK* sdk = [OPGSDK new];
    NSError *error;
    
    if ([[self.panelField.text description] isEqualToString:@""]) {
        self.surveyList = [sdk getUserSurveyList:&error];// Getting list of Surveys assigned to user
    }
    else{
        self.surveyList = [sdk getUserSurveyListWithPanelID:self.panelField.text error:&error]; // Getting list of Surveys for particular panel
    }
    
    
}

#pragma mark - Segues Operations
-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"getUserSurvey"]) {
        // Get the SurveyList view
        SurveyTableViewController *viewController = [segue destinationViewController];
        viewController.surveyListArray = self.surveyList;
    }
    
}

#pragma mark - TextField delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField{
    
    [textField resignFirstResponder];
    return YES;
}

@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/GetSurveysViewController.swift
//
//  GetSurveysViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class GetSurveysViewController: UIViewController, UITextFieldDelegate
{

    @IBOutlet weak var panelIDField: UITextField!
    var panelID : UITextField!
    var surveyList : NSArray = []
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        //panelID.delegate=self
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        return true
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    @IBAction func getSUrveySwift(_ sender: AnyObject)
    {
        
        let sdk = OPGSDK()        // Creating OPGSDK instance
        
        if self.panelIDField.text == ""
        {
            do {
                self.surveyList = try sdk.getUserSurveyList() as NSArray        // Getting list of Surveys assigned to user
            }
            catch{
                print("User Survey not working")         /* @"Error Occured. Contact Support!" */
                
            }
        }
        else
        {
            do {
                self.surveyList = try sdk.getUserSurveyList(withPanelID: self.panelIDField.text) as NSArray        // Getting list of Surveys assigned to user
            }
            catch{
                print("User Survey not working")         /* @"Error Occured. Contact Support!" */
                
            }
        }
        
        
        
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?)
    {
        if(segue.identifier == "getUserSurvey")
        {
            let surveyListVC : SurveyTableViewController  = segue.destination as! SurveyTableViewController
            surveyListVC.surveyListArray = self.surveyList
        }
    }

}
